### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by jonnyalexbh**
## Summary
- Sign Up

## Trello Card
https://trello.com/c/jlFleLOa/2-sign-up

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+exports.storeUser = (req, res, next) => {
+  const user = {
+    firstName: req.body.first_name,
+    lastName: req.body.last_name,
+    email: req.body.email,
+    password: helpers.encryptPassword(req.body.password)
+  };
+  return servicesUser
+    .createUser(user)
+    .then(result => res.send(result))
```
Comment (jersonlopez) [inline app/controllers/users.js]: is the response missing status

File: app/errors.js [hunk 2]
```diff
@@ -11,3 +11,9 @@ exports.defaultError = message => internalError(message, exports.DEFAULT_ERROR);
 
 exports.ALBUMS_API_ERROR = 'albums_api_error';
 exports.albumsApiError = message => internalError(message, exports.ALBUMS_API_ERROR);
+
+exports.EMAIL_EXIST = 'email_exist';
+exports.emailExist = message => internalError(message, exports.EMAIL_EXIST);
```
Comment (jersonlopez) [inline app/errors.js]: the error's name must be include the word error, like: `emailExistError`

File: app/helpers.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const bcrypt = require('bcryptjs'),
+  salt_sync = require('../config').encryption;
+
+const salt = () => bcrypt.genSaltSync(parseInt(salt_sync));
```
Comment (jersonlopez) [inline app/helpers.js]: unnecessary function, `bcrypt.genSaltSync` is already a function

`const salt = bcrypt.genSaltSync(parseInt(salt_sync));`

File: app/middlewares/errors.js [hunk 4]
```diff
@@ -6,7 +6,8 @@ const DEFAULT_STATUS_CODE = 500;
 const statusCodes = {
   [errors.DATABASE_ERROR]: 503,
   [errors.DEFAULT_ERROR]: 500,
-  [errors.ALBUMS_API_ERROR]: 503
+  [errors.ALBUMS_API_ERROR]: 503,
+  [errors.VALIDATION_FAILED]: 422
```
Comment (jersonlopez) [inline app/middlewares/errors.js]: i think that is a code 400 (bad request), what do you think?

File: app/middlewares/errors.js [hunk 5]
```diff
@@ -6,7 +6,8 @@ const DEFAULT_STATUS_CODE = 500;
 const statusCodes = {
```
Comment (jersonlopez) [inline app/middlewares/errors.js]: where is the EMAIL_EXIST code?

File: app/middlewares/validator.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
+const { check, validationResult } = require('express-validator/check'),
+  errors = require('../errors');
+
+const validateErrors = (req, res, next) => {
+  const validationErrors = validationResult(req);
+  if (!validationResult(req).isEmpty()) {
+    throw errors.validationFailed(validationErrors.array());
```
Comment (jersonlopez) [inline app/middlewares/validator.js]: I have the doubt if this throw works, could you please check.

File: app/helpers.js (hunk 3 above)
Comment (jonnyalexbh) [reply app/helpers.js]: it is necessary because it is the value that is passed to the genSaltSync

File: app/middlewares/validator.js (hunk 6 above)
Comment (jonnyalexbh) [reply app/middlewares/validator.js]: if it works it returns it this way
```suggestion
     "message": [
        {
            "value": "jonny.berrio@gmail.com",
            "msg": "The email does not belong to a Wolox domain.",
            "param": "email",
            "location": "body"
        }
    ],
    "internal_code": "validation_error"
}
```

**Pull request merged**